pkg/controllers/scheduler/scheduler.go

```diff
@@ -361,6 +361,39 @@ func (s *Scheduler) prepareToSchedule(
 ) {
 	logger := klog.FromContext(ctx)
 
+	// check migration config
+
+	existingMigrationConfigBytes, migrationConfigExists := fedObject.GetAnnotations()[common.MigrationConfigurationAnnotation]
+	_, customMigrationInfoExists := fedObject.GetAnnotations()[common.AppliedMigrationConfigurationAnnotation]
+	if !migrationConfigExists {
+		if customMigrationInfoExists {
+			delete(fedObject.GetAnnotations(), common.AppliedMigrationConfigurationAnnotation)
+		}
+	}
+	if migrationConfigExists {
+		customMigrationInfoBytes, err := getCustomMigrationInfoBytes(existingMigrationConfigBytes, fedObject, ftc)
+		if err == nil {
```

Comment:
It is idiomatic in golang to handle the `err != nil` case first. Is there any particular reason the `err == nil` case must be checked first?

Reply:
No particular reason. I'd like to change the order.